buffer/src/main/java/io/netty/buffer/UnpooledDirectByteBuf.java

```diff
@@ -482,36 +484,32 @@ public ByteBuf setBytes(int index, ByteBuffer src) {
 
     @Override
     public ByteBuf getBytes(int index, OutputStream out, int length) throws IOException {
-        getBytes(index, out, length, false);
-        return this;
-    }
-
-    private void getBytes(int index, OutputStream out, int length, boolean internal) throws IOException {
         ensureAccessible();
         if (length == 0) {
-            return;
+            return this;
         }
 
         if (buffer.hasArray()) {
             out.write(buffer.array(), index + buffer.arrayOffset(), length);
         } else {
-            byte[] tmp = new byte[length];
-            ByteBuffer tmpBuf;
-            if (internal) {
-                tmpBuf = internalNioBuffer();
-            } else {
-                tmpBuf = buffer.duplicate();
+            byte[] tmp = new byte[Math.min(length, BUFFER_SIZE)];
```

Comment:
did not use alloc() here because it is unclear for me if array().arrayOffset() returns non 0.

ByteBuffer.get() cannot work with a destination offset.